api/src/main/java/org/apache/iceberg/types/Conversions.java

```diff
@@ -117,6 +117,9 @@ public static ByteBuffer toByteBuffer(Type.TypeID typeId, Object value) {
         return (ByteBuffer) value;
       case DECIMAL:
         return ByteBuffer.wrap(((BigDecimal) value).unscaledValue().toByteArray());
+      case UNKNOWN:
```

Comment:
the lower/upper bound is converted to a byte buffer in https://github.com/apache/iceberg/blob/c07f2aabc0a1d02f068ecf1514d2479c0fbdd3b0/core/src/main/java/org/apache/iceberg/PartitionSummary.java#L80-L81